tips/TIP-0040/tip-0040.md

```diff
@@ -0,0 +1,733 @@
+
+Stakers, or interchangeably called _registered validators_, must lock their IOTA coins to stake them. By locking the
+tokens, the stakers prove control over them, so the protocol assigns them a proportional amount of _stake_ in exchange.
+Other users - the so-called delegators - can _delegate_ their IOTA coins to a staker to further increase their influence
+in the system. A staker plus all its delegators are called a _staking or validator pool_. Stakers are eligible to be
+selected into a committee for a certain duration. Only the votes of those selected into the committee are considered to
+determine the _Witness Weight_, used for the acceptance of blocks, and _Approval Weight_, used for conflict resolution
+and transaction acceptance. Stake is only used to determine who is selected for the committee. Once selected, every
+validator in the committee occupies a seat, and every seat has the same voting weight.
+
+The IOTA protocol slices time into slots and epochs. An epoch is simply a specified number of slots, and a slot has a
+protocol-defined duration in seconds. A new validator committee is selected for each epoch. Shortly before an epoch
+ends, the current activity of all registered validators, as well as the pools' stakes, are determined. Then, a committee
+is selected out of all the registered stakers with enough block issuance activity, depending on their pool stake. Once
+selected, each validator is responsible for issuing _Validation Blocks_ across the epoch through which they vote on
+conflicts. A performance factor is calculated based on how many such blocks they issued for each slot and how many they
+were expected to issue. This factor determines the amount of _Mana rewards_ the validator's pool receives.
+
+After an epoch ends, the delegators of the pool can claim their rewards. The validator may continue to stake or end its
+stake by going through an unbonding period of its locked tokens. After this period ends, the validator can unlock their
+IOTA coins and claim their Mana rewards.
+
+## Registration
+
+Accounts are considered registered for validation if they have a _Staking Feature_ in their _Features_. This feature and
+its transaction validation rules are specified in [TIP-42](../TIP-0042/tip-0042.md).
+
+## Committee Selection
+
+![](./assets/selection-timing.png)
+
+_This figure gives an overview of the timing around committee selection in epochs. Note that the exact length of those
+windows is not accurately depicted here and instead depends on the value of the corresponding Protocol Parameters._
+
+The following explains how the committee selection for a certain epoch `n` works, with the following time boundaries
+used in the algorithm:
+
+- Let `Epoch Start Slot(n))` be the first slot index of epoch `n`.
+- Let `Registration Slot(n)` be the registration slot with index
+  `Epoch Start Slot(n) - Epoch Nearing Threshold - Activity Window Duration - 1`. To be eligible for the committee
```

Comment:
What if this underflows?

Reply:
Just updated the formula to how it is in the code. 
Plus the protocol parameters have sanity checks now, which ensure that, for instance, "Epoch duration in slots must be strictly greater than EpochNearingThreshold", so EpochStartSlot - 1 (which is EpochEndSlot of the previous epoch) minus the nearing threshold can never underflow.

Reply:
Should these checks be part of https://github.com/iotaledger/tips/blob/tip49/tips/TIP-0049/tip-0049.md ?

Reply:
Good question. I'll ask in the team.

Reply:
Any update on this?

Reply:
Yes: We want to add it, but haven't done so yet.

Reply:
Added a task for it: https://github.com/iotaledger/tips/issues/161.